### ISSUE TITLE
Bug: Visitor keys for `ExportSpecifier` are not in source code order

### ISSUE THREAD

**Issue by overlookmotel** [typescript-eslint/typescript-eslint#11293]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

visitor-keys

### Repro Code

```TypeScript
export { local as exported }
```

### ESLint Config

```javascript

```

### tsconfig

```jsonc

```

### Expected Result

Visitation order for `ExportSpecifier` is incorrect - `exported` is visited before `local`.

`@typescript-eslint/visitor-keys` inherits this problem from `eslint-visitor-keys` (as discussed in #11279). It has been fixed in https://github.com/eslint/js/issues/655, which is included in `eslint-visitor-keys` v4.2.1, which was released today.

As far as I can see, this is the only substantive change made in v4.2.1, so I think it should be possible to bump `eslint-visitor-keys` dependency without upgrading to latest `eslint`, and it shouldn't affect anything else.

### Actual Result

n/a

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/visitor-keys`     | `8.34.0` |
